File: 1677-matrix-diagonal-sum/matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int n=mat.length;
        int sum=0;
        for(int i=0;i<n;i++){
            for(int j=0;j<mat[i].length;j++){
                if(n%2==0){
                    if(i==j){
                        sum+=mat[i][j];
                    }
                    else if(i+j==n-1){
                        sum+=mat[i][j];
                    }
                }
                else{
                    if(i==j){
                        sum+=mat[i][j];
                    }
                    else if(i+j==n-1){
                        sum+=mat[i][j];
                    }
                }
            }
        }
        return sum;
        
    }
}

File: 1677-matrix-diagonal-sum/matrix-diagonal-sum.py
class Solution(object):
    def diagonalSum(self, mat):
       sum=0;
       n=len(mat)
       for i in range(0,len(mat)):
        for j in range(0,len(mat[i])):
            if(i==j):
                sum+=mat[i][j]
            elif(i+j==n-1):
                sum+=mat[i][j]
       return sum
            
        